utils/S3UtilsMongoClient.js

```diff
@@ -0,0 +1,246 @@
+            location: {},
+            // there will be only one account for an object entry
+            account: { [entry.value['owner-display-name']]: size },
+        };
+
+        if (!isTransient
+            || entry.value.replicationInfo.status !== 'COMPLETED') {
+            // only count it in current dataStore if object is not in transient or replication not completed
+            if (results.location[entry.value.dataStoreName]) {
+                results.location[entry.value.dataStoreName] += size;
+            } else {
+                results.location[entry.value.dataStoreName] = size;
+            }
+        }
+        entry.value.replicationInfo.backends.forEach(rep => {
+            // count it in the replication destination location if replication compeleted
+            if (rep.status === 'COMPLETED') {
+                if (results.location[rep.site]) {
+                    results.location[rep.site] += size;
+                } else {
+                    results.location[rep.site] = size;
+                }
+            }
+        });
+        return {
+            data: results,
+            error: null,
+        };
+    }
+
+    _handleResults(res, isVersioned) {
+        let totalNonCurrentCount = 0;
+        let totalCurrentCount = 0;
+        const totalBytes = { curr: 0, prev: 0 };
+        const locationBytes = {};
+        const dataMetrics = {
+            bucket: {},
+            location: {},
+            account: {},
+        };
+
+        Object.keys(res).forEach(metricLevel => {
+            // metricLevel can only be 'bucket', 'location' or 'account'
+            if (validStorageMetricLevels.has(metricLevel)) {
+                Object.keys(res[metricLevel]).forEach(resource => {
+                    // resource can be the name of bucket, location or account
+                    const resourceName = metricLevel === 'location' ? this._getLocName(resource) : resource;
+                    if (!dataMetrics[metricLevel][resourceName]) {
+                        dataMetrics[metricLevel][resourceName] = {
+                            usedCapacity: {
+                                current: 0,
+                                nonCurrent: 0,
+                            },
+                            objectCount: {
+                                current: 0,
+                                nonCurrent: 0,
+                                deleteMarker: 0,
+                            },
+                        };
+                    }
+                    const {
+                        masterCount,
+                        masterData,
+                        nullCount,
+                        nullData,
+                        versionCount,
+                        versionData,
+                        deleteMarkerCount,
+                    } = res[metricLevel][resourceName];
+
+                    dataMetrics[metricLevel][resourceName].usedCapacity.current += nullData + masterData;
```

Comment:
`nullData` is referencing the size of the objects in an unversioned bucket.
`masterData` is referencing the size of the master versions in a versioned/suspended bucket.

For a given bucket state one or the other shoule be 0. Is it why we add them here?

Reply:
yeah, `nullData` represent current data in nonversioned bucket/ object uploaded after bucket suspended.
`masterData` represents current data in versioned bucket/object uploaded before bucket suspended
if one has value, the other is 0